pkg/trace/api/profiles.go

```diff
@@ -22,60 +25,141 @@ const (
 	profilingURLDefault = "https://intake.profile.datadoghq.com/v1/input"
 )
 
-// profilingEndpoint returns the profiling intake API URL based on agent configuration.
-func profilingEndpoint() string {
+// profilingEndpoints returns the profiling intake urls and their corresponding
+// api keys based on agent configuration. The main endpoint is always returned as
+// the first element in the slice.
+func profilingEndpoints(apiKey string) (urls []*url.URL, apiKeys []string, err error) {
+	main := profilingURLDefault
 	if v := config.Datadog.GetString("apm_config.profiling_dd_url"); v != "" {
-		return v
+		main = v
+	} else if site := config.Datadog.GetString("site"); site != "" {
+		main = fmt.Sprintf(profilingURLTemplate, site)
 	}
-	if site := config.Datadog.GetString("site"); site != "" {
-		return fmt.Sprintf(profilingURLTemplate, site)
+	u, err := url.Parse(main)
+	if err != nil {
+		// if the main intake URL is invalid we don't use additional endpoints
+		return nil, nil, fmt.Errorf("error parsing main profiling intake URL %s", main)
```

Comment:
```suggestion
		return nil, nil, fmt.Errorf("error parsing main profiling intake URL %s: %v", main, err)
```
We should know what went wrong, it can help on occasion.